### PR TITLE
COMMERCE-12218

### PR DESCRIPTION
**Pull request by Brunno0liveira**
https://liferay.atlassian.net/browse/COMMERCE-12233

### CODE REVIEW

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 1]
```diff
@@ -158,6 +158,223 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-12218. Admin can added tier price on a product's SKU details page"
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:161]: This is a test for COMMERCE-12218. Admin can add tier price on a product's SKU details page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 2]
```diff
@@ -158,6 +158,223 @@ definition {
...
+
+			CommerceEntry.gotoMenuTab(menuTab = "Price Lists");
+
+			CommerceNavigator.gotoEntry(entryName = "Minium Base Price List");
+
+			CommercePriceLists.createTierPriceEntryDirectlyInsideProductSKU(
+				price = 10,
+				quantity = 1);
+
+			Refresh();
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:299]: If possible use SelectFrameTop

Thanks!

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 3]
```diff
@@ -158,6 +158,223 @@ definition {
...
+				locator1 = "Panel#PANEL_COLLAPSED");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Tier Price Entries");
+
+			CommerceEntry.clickEntryDropdownMenuItem(
+				entryName = 10,
+				menuItem = "Delete");
+		}
+
+		task ("Then the tier price is not longer visible and the price goes back to the base price list") {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:324]: Then the tier price is not applied anymore

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 4]
```diff
@@ -158,6 +158,223 @@ definition {
...
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			AssertTextEquals.assertPartialText(
+				key_productName = "Test Simple Product",
+				locator1 = "CommerceEntry#PRODUCT_CARD_LIST_PRICE",
+				value1 = 00);
+		}
+	}
+
+	@description = "This is a test for COMMERCE-12218. Admin can edited price lists on a product's SKU details page"
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:334]: Admin can edit price lists on a product's SKU details page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 5]
```diff
@@ -158,6 +158,223 @@ definition {
...
+
+		task ("And a simple product with 'Minium' catalog is created") {
+			CommerceJSONProductsAPI._addCommerceProduct(
+				catalogName = "Minium",
+				productName = "Test Simple Product",
+				productType = "Simple",
+				sku = "SIMPLESKU");
+		}
+
+		task ("When admin edited a price list on the product's SKU details page") {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:351]: When admin edit a price list on the product's SKU details page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 6]
```diff
@@ -237,6 +454,74 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-12218. Admin can edited tier price on a product's SKU details page"
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:457]: Admin can edit tier price on a product's SKU details page

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 7]
```diff
@@ -384,6 +669,101 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-12218. Admin can view channel currency matches catalogue currency but dosen't change the price list on a product's SKU details page"
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:672]: Admin can view channel currency matches catalog currency but **doesn't** change the price list on a product's SKU details page

**Review by pietrobocale — CHANGES_REQUESTED** [review]
Review Completed, see the comments

Thanks!

**Comment by pietrobocale** [comment]
It is better to add another scenario similar to CanViewChannelCurrencyDoesntChangePriceListOnProductSKU. The difference is that in the new scenario, we will change the currency inside the Price List so that the currency inside the SKU Price List will change.

Thanks!

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase [hunk 8]
```diff
@@ -384,6 +671,180 @@ definition {
...
+
+			Select(
+				key_fieldLabel = "Currency",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "EUR");
+
+			Button.clickPublish();
+		}
+
+		task ("Then the currency updated on the the price list inside the product's SKU details page ") {
```
Comment (pietrobocale) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSKUs.testcase:823]: Then the currency is updated inside the product's SKU details page

**Review by pietrobocale — CHANGES_REQUESTED** [review]
Review Compleated, check the comments

Thanks!

**Comment by pietrobocale** [comment]
@Brunno0liveira Can you rename the commit "Add Test Coverage of SKU price List Setting" into "Add new functional tests"?

Thanks!

**Comment by Brunno0liveira** [comment]
ci:test:sf

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Review by pietrobocale — APPROVED** [review]
Review Passed

Thanks!

**Comment by Brunno0liveira** [comment]
ci:forward

**Pull request closed without merging**